app/scripts/lib/ppom/ppom-util.ts

```diff
@@ -178,3 +173,35 @@ async function findConfirmationBySecurityAlertId(
     await new Promise((resolve) => setTimeout(resolve, 100));
   }
 }
+
+async function validateWithController(
+  ppomController: PPOMController,
+  request: JsonRpcRequest,
+): Promise<SecurityAlertResponse> {
+  const response = (await ppomController.usePPOM((ppom: PPOM) =>
+    ppom.validateJsonRpc(request),
+  )) as SecurityAlertResponse;
+
+  return {
+    ...response,
+    source: SecurityAlertSource.Local,
```

Comment:
We're adding this to the response, but do we also have to update `getBlockaidMetricsProps` in `ui/helpers/utils/metrics.js` to add it to the event?